src/codegen/mod.rs

```diff
@@ -1342,3 +1344,25 @@ impl From<&ast::Builtin> for Builtin {
         }
     }
 }
+
+pub(super) fn print_runtime_error(ns: &Namespace, error: &str, loc: Option<Loc>) -> String {
```

Comment:
The function name is confusing, as it doesn't have anything to do with "runtime error printing"; additionally there is now already a `fn log_runtime_error` which adds up confusion. Can you please  rename it to `error_msg_with_loc` or something like that?

Reply:
Good point, I think ``format_error_msg_with_loc`` is nice